src/web/extractors.rs

```diff
@@ -120,55 +150,67 @@ impl FromRequest<ServerState> for BsoBodies {
                     // Skip any blanks
                     if item == "" {
                         continue;
                     }
-                    if let Ok(item) = serde_json::from_str(&item) {
-                        bsos.push(item);
+                    // Check that its a valid JSON map like we expect
+                    if let Ok(raw_json) = serde_json::from_str::<Value>(&item) {
+                        bsos.push(raw_json);
                     } else {
-                        // Per Python version, all BSO's must parse or we error out
+                        // Per Python version, BSO's must json deserialize
                         return future::err(make_error());
                     }
                 }
                 bsos
             } else {
-                if let Ok(bsos) = serde_json::from_str(&body) {
-                    bsos
+                if let Ok(json_vals) = serde_json::from_str::<Vec<Value>>(&body) {
+                    json_vals
                 } else {
+                    // Per Python version, BSO's must json deserialize
                     return future::err(make_error());
                 }
             };
 
             // Validate all the BSO's, move invalid to our other list. Assume they'll all make
             // it with our pre-allocation
             let mut valid: Vec<BatchBsoBody> = Vec::with_capacity(bsos.len());
-            let mut invalid: Vec<BsoBody> = Vec::new();
-            for mut bso in bsos {
-                if bso.validate().is_ok() {
-                    // They're only valid if they include an id.
-                    // XXX: Verify the id is optional, this may be a legacy thing
-                    match bso {
-                        BsoBody {
-                            id: Some(id),
-                            sortindex,
-                            payload,
-                            ttl,
-                        } => valid.push(BatchBsoBody {
-                            id,
-                            sortindex,
-                            payload,
-                            ttl,
-                        }),
-                        bso => invalid.push(bso),
+
+            // Invalid BSO's are any BSO that can deserialize despite how wrong the contents are
+            // per the way the Python version works.
+            let mut invalid: Vec<Value> = Vec::new();
+
+            // Keep track of our total payload size
+            let mut total_payload_size = 0;
+
+            for bso in bsos {
+                // Error out if its not a JSON mapping type
+                if !bso.is_object() {
+                    return future::err(make_error());
+                }
+                match BatchBsoBody::from_raw_bso(&bso) {
+                    Some(b) => {
+                        // Is this record too large? Deny if it is.
+                        let payload_size = b.payload.clone().map(|v| v.len()).unwrap_or_default();
```

Comment:
avoid cloning these 'larger' payload chunks
```suggestion
                        let payload_size = b.payload.as_ref().map(|v| v.len()).unwrap_or_default();
```